src/three/PNTSLoader.js

```diff
@@ -1,53 +1,84 @@
 import { PNTSLoaderBase } from '../base/PNTSLoaderBase.js';
-import { Points, PointsMaterial, BufferGeometry, BufferAttribute, DefaultLoadingManager } from 'three';
-
+import {
+	Points,
+	PointsMaterial,
+	BufferGeometry,
+	BufferAttribute,
+	DefaultLoadingManager,
+} from 'three';
+
+const DRACO_ATTRIBUTE_MAP = {
+	RGB: 'color',
+	POSITION: 'position',
+};
 export class PNTSLoader extends PNTSLoaderBase {
 
 	constructor( manager = DefaultLoadingManager ) {
 
 		super();
 		this.manager = manager;
 
+		// hacky way of getting the draco loader from the manager
+		this.dracoLoader = this.manager.getHandler( 'draco.drc' );
+
 	}
 
 	parse( buffer ) {
 
-		return super
-			.parse( buffer )
-			.then( result => {
+		return super.parse( buffer ).then( async ( result ) => {
 
-				const { featureTable } = result;
+			const { featureTable } = result;
 
-				const POINTS_LENGTH = featureTable.getData( 'POINTS_LENGTH' );
-				const POSITION = featureTable.getData( 'POSITION', POINTS_LENGTH, 'FLOAT', 'VEC3' );
-				const RGB = featureTable.getData( 'RGB', POINTS_LENGTH, 'UNSIGNED_BYTE', 'VEC3' );
+			const geometry = new BufferGeometry();
+			const material = new PointsMaterial();
 
-				[
-					'QUANTIZED_VOLUME_OFFSET',
-					'QUANTIZED_VOLUME_SCALE',
-					'CONSTANT_RGBA',
-					'BATCH_LENGTH',
-					'POSITION_QUANTIZED',
-					'RGBA',
-					'RGB565',
-					'NORMAL',
-					'NORMAL_OCT16P',
-				].forEach( feature => {
+			if ( featureTable.isDraco() ) {
 
-					if ( feature in featureTable.header ) {
+				const dracoIDs = featureTable.getDracoProperties();
+				const attributeIDs = {};
 
-						console.warn( `PNTSLoader: Unsupported FeatureTable feature "${ feature }" detected.` );
+				for ( const [ key, value ] of Object.entries( dracoIDs ) ) {
 
-					}
+					attributeIDs[ DRACO_ATTRIBUTE_MAP[ key ] ] = value;
 
-				} );
+				}
 
-				const geometry = new BufferGeometry();
-				geometry.setAttribute( 'position', new BufferAttribute( POSITION, 3, false ) );
+				const taskConfig = {
+					attributeIDs,
+					attributeTypes: {
+						position: 'Float32Array',
+						color: 'Uint8Array',
+					},
+					useUniqueIDs: true,
+				};
+
+				const buffer = featureTable.getDracoBuffer();
+
+				if ( this.dracoLoader == null ) {
 
-				const material = new PointsMaterial();
-				material.size = 2;
-				material.sizeAttenuation = false;
+					throw new Error( 'PNTSLoader: dracoLoader not available.' );
+
+				}
+
+				const dracoGeometry = await this.dracoLoader
+					.decodeGeometry( buffer, taskConfig );
+
+				geometry.copy( dracoGeometry );
+
+				if ( geometry.attributes.color ) {
+
+					geometry.attributes.color.normalized = true;
```

Comment:
Does the DRACO loader not set the color field to normalized correctly automatically? Is this something that should be fixed in three.js?

Reply:
Good idea, I'll have a look

Reply:
made an issue and PR [here](https://github.com/mrdoob/three.js/pull/26036)
unsure if this desired behaviour for everyone, but feels like it